### PR TITLE
High-order Nodal Fields and Solution Transfer

### PR DESCRIPTION
**Pull request by mortezah**
This pull request achieves the following

- adds higher order nodal (hierarchic) fields similar to the implementation of MFEM
- adds a regression test for such fields
- updates the solution transfer codes so that these higher order fields are also transferred during adapt 
- adds a regression test for higher order solutions transfer

### CODE REVIEW

**Comment by cwsmith** [comment]
Looks good at first glance.  I'll take a closer look tomorrow. 

Please add any new source files to the `pkg_tribits.cmake` files in each directory.

**Comment by mortezah** [comment]
@cwsmith the new source are in the pkg_tribits.cmake

**Comment by mortezah** [comment]
@cwsmith 

Do you know when this would be merged? A student I am working with will need these changes for his development.

**Comment by cwsmith** [comment]
Sorry for the delay.  I'll look in about an hour.

File: apf/apfH1Shapes.cc
Comment (cwsmith) [inline apf/apfH1Shapes.cc:267]: Is there a tab here?  The indentation looks odd.

File: apf/apfH1Shapes.cc
Comment (cwsmith) [inline apf/apfH1Shapes.cc:21]: Is this needed?  I don't see much stl usage below.

File: apf/apfH1Shapes.cc
Comment (cwsmith) [inline apf/apfH1Shapes.cc:327]: tabs?  github does not always render the text white space correctly

File: apf/apfH1Shapes.cc
Comment (cwsmith) [inline apf/apfH1Shapes.cc:406]: tabs?

File: apf/apfH1Shapes.cc
Comment (cwsmith) [inline apf/apfH1Shapes.cc:465]: tabs?

File: ma/maSolutionTransfer.cc
Comment (cwsmith) [inline ma/maSolutionTransfer.cc:135]: the indentation looks odd here - it could be github

File: test/H1Shapes.cc
Comment (cwsmith) [inline test/H1Shapes.cc:18]: is this needed?

File: test/testing.cmake
Comment (cwsmith) [inline test/testing.cmake:531]: any reason not to load the model? (same for the following test)

**Review by cwsmith — APPROVED** [review]
I didn't see any obvious major issues.  

There are a few spots to check for indentation.  

The use of `using namespace std` at the top of a source (and definitely a header) is generally not great since it 'infects' the entire file with the contents of `std` namespace.   

Marking approved so this can be merged once the small edits are made.

File: apf/apfH1Shapes.cc
Comment (mortezah) [reply apf/apfH1Shapes.cc]: Fixed this.

File: apf/apfH1Shapes.cc
Comment (mortezah) [reply apf/apfH1Shapes.cc]: Fixed this.

File: apf/apfH1Shapes.cc
Comment (mortezah) [reply apf/apfH1Shapes.cc]: Fixed this.

File: apf/apfH1Shapes.cc
Comment (mortezah) [reply apf/apfH1Shapes.cc]: Fixed this.

File: ma/maSolutionTransfer.cc
Comment (mortezah) [reply ma/maSolutionTransfer.cc:135]: Fixed this.

File: apf/apfH1Shapes.cc
Comment (mortezah) [reply apf/apfH1Shapes.cc]: Fixed this.

File: test/H1Shapes.cc
Comment (mortezah) [reply test/H1Shapes.cc]: Fixed this.

File: test/testing.cmake
Comment (mortezah) [reply test/testing.cmake]: These test only depend on the mesh. I have added the models to keep things uniform with the surrounding code.

**Comment by mortezah** [comment]
@cwsmith thanks for the reviews. I am going to merge this shortly if there are no more issues. Github has a few options

* merge commit
* squash and merge
* rebase and merge

Which one is the suggested method?

**Comment by cwsmith** [comment]
You're welcome.  

The answer depends on what history (sequence of commits) you want to exist in develop after the merge.

- `merge commit`will add the sequence of commits from the PR to develop + a 'merge' commit. 
- `squash and merge` will combine ('squash') all the commits from this PR into a new commit and add it and a 'merge' commit into develop.
- `rebase and merge` will squash the commits and add that commit onto develop

This appears like a decent, deeper, explanation of the differences:
https://rietta.com/blog/github-merge-types/

**Comment by mortezah** [comment]
@cwsmith thanks.

I will do the merge commit, since I will need to refer to some of the commits (for code style and development purposes) when I work with the students.

**Pull request merged**
